### PR TITLE
Use native OS build workers

### PR DESCRIPTION
**Pull request by smlx**
This PR updates the Github build and release workflows to be OS-native: macOS binaries built on macOS, and Linux binaries built on Linux.

It also updates the `build` target in the `Makefile` to handle the split `goreleaser` configuration.

It should fix the issue described in https://github.com/Versent/saml2aws/pull/961

### CODE REVIEW

**Comment by smlx** [comment]
cc @mapkon

**Review by mapkon — CHANGES_REQUESTED** [review]
Thank you for the PR - but I don't think it is ready yet.

It looks like the split is not working properly - please see the output from a scratch repo I created here: https://github.com/mapkon/go-releaser-tinker/actions/runs/4319873504/jobs/7539569017

It appears that the second release (macos-latest) attempts to (re)upload the `go-releaser-tinker_0.1.0_checksums.txt` file, leading to this error:

```
  ⨯ release failed after 12s                 
error=scm releases: failed to publish artifacts: failed to upload go-releaser-tinker_0.1.0_checksums.txt after 1 tries: 
POST https://uploads.github.com/repos/mapkon/go-releaser-tinker/releases/94362976/assets?name=go-releaser-tinker_0.1.0_checksums.txt: 422 Validation Failed 
[{Resource:ReleaseAsset Field:name Code:already_exists Message:}]

Error: The process '/Users/runner/hostedtoolcache/goreleaser-action/1.15.2/x64/goreleaser' failed with exit code 1
```

Can you take a look?

File: .goreleaser.ubuntu-latest.yml [hunk 1]
```diff
@@ -34,3 +34,5 @@ archives:
     files:
       - LICENSE.md
       - README.md
+checksum:
+  name_template: "{{ .ProjectName }}_{{ .Version }}_linux_windows_checksums.txt"
```
Comment (mapkon) [inline .goreleaser.ubuntu-latest.yml:38]: Would it make sense to just use the `"{{ .ProjectName }}_{{ .Version }}_checksums.txt"` to cater for Windows and Linux?

CC: @wolfeidau

Comment (smlx) [reply .goreleaser.ubuntu-latest.yml]: ok, I've made the change

**Comment by smlx** [comment]
Sorry about the bug - I've tested a few times and should be fixed now.

**Comment by mapkon** [comment]
> Sorry about the bug - I've tested a few times and should be fixed now.

No worries! Will take a look at the earliest opportune time.

**Pull request merged**